rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp

```diff
@@ -191,13 +194,25 @@ void SequentialReader::set_filter(
 
 void SequentialReader::reset_filter()
 {
-  topics_filter_ = rosbag2_storage::StorageFilter();
+  set_filter(rosbag2_storage::StorageFilter());
+}
+
+void SequentialReader::seek(const rcutils_time_point_value_t & timestamp)
+{
+  seek_time_ = timestamp;
   if (storage_) {
-    storage_->reset_filter();
+    // reset to the first file
+    current_file_iterator_ = file_paths_.begin();
+    preprocess_current_file();
+    storage_options_.uri = get_current_file();
+    // open and set filters
+    storage_ = storage_factory_->open_read_only(storage_options_);
```

Comment:
Is there much overhead to this opening? We don't have to block first pass on this consideration, but I might put a TODO here to do something more performant - unless it's not a problem or there's a strong reason we _have_ to reload the storage.

Reply:
are you asking why we need to reopen files here? there's no guarantee that the seek time is in the current file. if we have per-file metadata around the start/end time of each file, i guess we can just set to the first file that satisfies. i'm not sure why we're particularly concerned about this reopen as it happens once per seek. we already do it on every roll over. It's true that if the bag is split into many files the seek + subsequent rollovers before hitting the first message might add up.

Reply:
My concern was more around repeated seeks within a single file, such as in a "video scrubber" GUI interface - however we can leave this consideration for a real use case and apply performance testing rather than having me speculate on it. Let's move forward with this implementation.